### PR TITLE
Clean up hlint warnings

### PR DESCRIPTION
**Pull request by SanchayanMaity**
Some more cleanups for #406 

What benchmarks should I run to test for regression here? This touches a lot of the internal code.

Signed-off-by: Sanchayan Maity <maitysanchayan@gmail.com>

### CODE REVIEW

File: src/Streamly/Internal/Data/Stream/Enumeration.hs [hunk 1]
```diff
@@ -298,9 +298,10 @@ enumerateFromThenToSmall from next to = Serial.map toEnum $
 enumerateFromThenSmallBounded :: (IsStream t, Monad m, Enumerable a, Bounded a)
     => a -> a -> t m a
 enumerateFromThenSmallBounded from next =
-    case fromEnum next >= fromEnum from of
-        True -> enumerateFromThenTo from next maxBound
-        False -> enumerateFromThenTo from next minBound
+    if fromEnum next >= fromEnum from then
+        enumerateFromThenTo from next maxBound
+    else
+        enumerateFromThenTo from next minBound
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Stream/Enumeration.hs:304]: We format if-then-else as follows:

```
if fromEnum next >= fromEnum from
then enumerateFromThenTo from next maxBound
else enumerateFromThenTo from next minBound
```

File: src/Streamly/Internal/Data/Stream/StreamD/Type.hs [hunk 2]
```diff
@@ -1,3 +1,5 @@
+{-# LANGUAGE LambdaCase #-}
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Stream/StreamD/Type.hs:1]: LambdaCase is already enabled in default-extensions.

**Comment by harendra-kumar** [comment]
You can run stream benchmarks. But you can also run them all with the `--quick` flag (`bench.sh` script) it does not take too long. Watch for difference in allocations.

File: src/Streamly/Internal/Data/Stream/Instances.hs [hunk 3]
```diff
@@ -142,15 +142,15 @@ instance (Foldable m, Monad m) => Foldable (STREAM m) where {                 \
                                                                               \
     {-# INLINE maximum #-};                                                   \
     maximum =                                                                 \
-          fromMaybe (errorWithoutStackTrace $ "maximum: empty stream")        \
+          fromMaybe (errorWithoutStackTrace "maximum: empty stream")        \
```
Comment (adithyaov) [inline src/Streamly/Internal/Data/Stream/Instances.hs:145]: Not a big deal, but maybe align `\` with the rest of the lines?

File: src/Streamly/Internal/Data/Stream/Instances.hs [hunk 4]
```diff
@@ -142,15 +142,15 @@ instance (Foldable m, Monad m) => Foldable (STREAM m) where {                 \
...
         . toMaybe                                                             \
         . foldl' getMax Nothing' where {                                      \
             getMax Nothing' x = Just' x;                                      \
             getMax (Just' mx) x = Just' $! max mx x };                        \
                                                                               \
     {-# INLINE minimum #-};                                                   \
     minimum =                                                                 \
-          fromMaybe (errorWithoutStackTrace $ "minimum: empty stream")        \
+          fromMaybe (errorWithoutStackTrace "minimum: empty stream")        \
```
Comment (adithyaov) [inline src/Streamly/Internal/Data/Stream/Instances.hs:153]: Align `\` with the rest of the lines?

**Comment by harendra-kumar** [comment]
@SanchayanMaity sorry to keep this waiting for too long. Can you please rebase and push it again?

**Comment by harendra-kumar** [comment]
Ah, I missed the new push on this. @SanchayanMaity you should have pinged after pushing. Anyway, I will resolve it myself and merge it.

**Comment by harendra-kumar** [comment]
Its not possible to rebase this, the changes will have to be patched which will make me the author.

**Comment by SanchayanMaity** [comment]
@harendra-kumar I can redo this sometime early next week if that is OK with you.

**Comment by harendra-kumar** [comment]
> @harendra-kumar I can redo this sometime early next week if that is OK with you.

Sure, thanks a lot.

**Comment by harendra-kumar** [comment]
Not a great timing as I have a large refactor pending. But I will merge this and take care of the conflicts.

**Pull request merged**